pulumi/src/statefulService.ts

```diff
@@ -335,9 +315,9 @@ export async function deployStatefulService(
 
     const additionalRootDomainName = process.env.ADDITIONAL_ROOT_DOMAIN_NAME
 
-    const match = (service: string, prefix?: string) => {
+    const match = (service: Service, prefix?: string) => {
```

Comment:
if you are going to pass the full service, you need to use `service.name` for all the string interpolation. Could just leave it as a string and pass in `service.name` in the map though makes a bit more sense to me

Reply:
> you need to use service.name
Isn't that the case right now?

Reply:
The latter decreases type safety, which in case of this specific function I'd rather avoid. I have a hard time navigating this 200 line context as is

Reply:
then we need `service.name` for the string interpolation. will make another comment